### PR TITLE
Remove datacatalog deps

### PR DESCRIPTION
**Pull request by tnsetting**
## Read then delete
- Make sure to use a concise title for the pull-request.
- Use #patch, #minor or #major in the pull-request title to bump the corresponding version. Otherwise, the patch version
will be bumped. [More details](https://github.com/marketplace/actions/github-tag-bump)

# TL;DR
Remove the dependency from datacatalog and use the gen code from flyte idl

## Type
 - [ ] Bug Fix
 - [ ] Feature
 - [ ] Plugin

## Are all requirements met?

 - [ ] Code completed
 - [ ] Smoke tested
 - [ ] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
 _How did you fix the bug, make the feature etc. Link to any design docs etc_

## Tracking Issue
https://github.com/flyteorg/flyte/issues/827

## Follow-up issue
_NA_
OR
_https://github.com/lyft/flyte/issues/<number>_

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -6,8 +6,7 @@ require (
 	github.com/DiSiqueira/GoTree v1.0.1-0.20180907134536-53a8e837f295
 	github.com/benlaurie/objecthash v0.0.0-20180202135721-d1e3d6079fc1
 	github.com/fatih/color v1.10.0
-	github.com/flyteorg/datacatalog v0.3.0
-	github.com/flyteorg/flyteidl v0.18.15
+	github.com/flyteorg/flyteidl v0.18.17
```
Comment (EngHabu) [inline go.mod:9]: I think we are up to 0.18.20 now... can you run `go get -u github.com/flyteorg/flyteidl`

File: go.sum [hunk 2]
```diff
@@ -239,6 +239,8 @@ github.com/flyteorg/datacatalog v0.3.0 h1:rkZ/IJTPGAJjHig2O+iSq9W9xqqVZdJTwGYb2Q
 github.com/flyteorg/datacatalog v0.3.0/go.mod h1:Y022upSReYrjYCO+S+/udfuVOBj047bxgJkA+2irHRI=
 github.com/flyteorg/flyteidl v0.18.15 h1:sXrlwTRaRjQsXYMNrY/S930SKdKtu4XnpNFEu8I4tn4=
 github.com/flyteorg/flyteidl v0.18.15/go.mod h1:b5Fq4Z8a5b0mF6pEwTd48ufvikUGVkWSjZiMT0ZtqKI=
+github.com/flyteorg/flyteidl v0.18.17 h1:74pPZ9PzITuzq+CgjMPb9EcFI5bVkf8mM5m4xmmlTmY=
```
Comment (EngHabu) [inline go.sum:242]: can you run `go mod tidy` so that it removes the older versions if they are no longer needed...

**Comment by kumare3** [comment]
Thank you so much, looks great to me

**Review by kumare3 — APPROVED** [review]

**Comment by kumare3** [comment]
I looked at the tests, how can they break?

**Comment by kumare3** [comment]
the test is a red-herring, weirdly there is a panic being raised
```panic when reconciling workflow, Stack: [goroutine 73 [running]:\nruntime/debug.Stack(0x3eda434, 0x1, 0x1)\n\t/usr/local/go/src/runtime/debug/stack.go:24 +0xab\ngithub.com/flyteorg/flytepropeller/pkg/controller.(*Propeller).TryMutateWorkflow.func2.1(0x2e21f20, 0xc000493900, 0xc0007f99c8, 0x2e6ece0, 0xc0003f5f50, 0xc00099e6c0)\n\t/go/src/github.com/flyteorg/flytepropeller/pkg/controller/handler.go:116 +0xbd\npanic(0x2574900, 0x2df4160)\n\t/usr/local/go/src/runtime/panic.go:975 +0x3e3\ngithub.com/flyteorg/flytepropeller/pkg/controller.TestPropeller_Handle.func10.1(0x2e6ece0, 0xc0003f5f50, 0xc0006fe480, 0x3f5a2e0, 0x465e96)\n\t/go/src/github.com/flyteorg/flytepropeller/pkg/controller/handler_test.go:274 +0x47\ngithub.com/flyteorg/flytepropeller/pkg/controller.(*mockExecutor).HandleFlyteWorkflow(0xc00067d170, 0x2e6ece0, 0xc0003f5f50, 0xc0006fe480, 0x0, 0x0)\n\t/go/src/github.com/flyteorg/flytepropeller/pkg/controller/handler_test.go:35 +0x69\ngithub.com/flyteorg/flytepropeller/pkg/controller.(*Propeller).TryMutateWorkflow.func2(0xc00099e6c0, 0x2e6ece0, 0xc0003f5f50, 0xc0007f99c8, 0xc0006fe480)\n\t/go/src/github.com/flyteorg/flytepropeller/pkg/controller/handler.go:122 +0x21a\ngithub.com/flyteorg/flytepropeller/pkg/controller.(*Propeller).TryMutateWorkflow(0xc00099e6c0, 0x2e6ec60, 0xc000048080, 0xc0006fe240, 0x0, 0x0, 0x0)\n\t/go/src/github.com/flyteorg/flytepropeller/pkg/controller/handler.go:123 +0x84b\ngithub.com/flyteorg/flytepropeller/pkg/controller.(*Propeller).Handle(0xc00099e6c0, 0x2e6ec60, 0xc000048080, 0x289c05c, 0x4, 0x289b15a, 0x3, 0x0, 0x0)\n\t/go/src/github.com/flyteorg/flytepropeller/pkg/controller/handler.go:197 +0xca0\ngithub.com/flyteorg/flytepropeller/pkg/controller.TestPropeller_Handle.func10(0xc0009598c0)\n\t/go/src/github.com/flyteorg/flytepropeller/pkg/controller/handler_test.go:276 +0x3f6\ntesting.tRunner(0xc0009598c0, 0xc0003f5dd0)\n\t/usr/local/go/src/testing/testing.go:1050 +0x1ec\ncreated by testing.(*T).Run\n\t/usr/local/go/src/testing/testing.go:1095 +0x538\n]" res_ver= wf=w1
time="2021-03-16T20:06:59Z" level=error msg="Error when trying to reconcile workflow. Error [panic when reconciling workflow, Stack: [goroutine 73 [running]:\nruntime/debug.Stack(0x3eda434, 0x1, 0x1)\n\t/usr/local/go/src/runtime/debug/stack.go:24 +0xab\ngithub.com/flyteorg/flytepropeller/pkg/controller.(*Propeller).TryMutateWorkflow.func2.1(0x2e21f20, 0xc000493900, 0xc0007f99c8, 0x2e6ece0, 0xc0003f5f50, 0xc00099e6c0)\n\t/go/src/github.com/flyteorg/flytepropeller/pkg/controller/handler.go:116 +0xbd\npanic(0x2574900, 0x2df4160)\n\t/usr/local/go/src/runtime/panic.go:975 +0x3e3\ngithub.com/flyteorg/flytepropeller/pkg/controller.TestPropeller_Handle.func10.1(0x2e6ece0, 0xc0003f5f50, 0xc0006fe480, 0x3f5a2e0, 0x465e96)\n\t/go/src/github.com/flyteorg/flytepropeller/pkg/controller/handler_test.go:274 +0x47\ngithub.com/flyteorg/flytepropeller/pkg/controller.(*mockExecutor).HandleFlyteWorkflow(0xc00067d170, 0x2e6ece0, 0xc0003f5f50, 0xc0006fe480, 0x0, 0x0)\n\t/go/src/github.com/flyteorg/flytepropeller/pkg/controller/handler_test.go:35 +0x69\ngithub.com/flyteorg/flytepropeller/pkg/controller.(*Propeller).TryMutateWorkflow.func2(0xc00099e6c0, 0x2e6ece0, 0xc0003f5f50, 0xc0007f99c8, 0xc0006fe480)\n\t/go/src/github.com/flyteorg/flytepropeller/pkg/controller/handler.go:122 +0x21a\ngithub.com/flyteorg/flytepropeller/pkg/controller.(*Propeller).TryMutateWorkflow(0xc00099e6c0, 0x2e6ec60, 0xc000048080, 0xc0006fe240, 0x0, 0x0, 0x0)\n\t/go/src/github.com/flyteorg/flytepropeller/pkg/controller/handler.go:123 +0x84b\ngithub.com/flyteorg/flytepropeller/pkg/controller.(*Propeller).Handle(0xc00099e6c0, 0x2e6ec60, 0xc000048080, 0x289c05c, 0x4, 0x289b15a, 0x3, 0x0, 0x0)\n\t/go/src/github.com/flyteorg/flytepropeller/pkg/controller/handler.go:197 +0xca0\ngithub.com/flyteorg/flytepropeller/pkg/controller.TestPropeller_Handle.func10(0xc0009598c0)\n\t/go/src/github.com/flyteorg/flytepropeller/pkg/controller/handler_test.go:276 +0x3f6\ntesting.tRunner(0xc0009598c0, 0xc0003f5dd0)\n\t/usr/local/go/src/testing/testing.go:1050 +0x1ec\ncreated by testing.(*T).Run\n\t/usr/local/go/src/testing/testing.go:1095 +0x538\n]]. Error Type[*errors.errorString]. Is nill [%!v(MISSING)]" res_ver= wf=w1
```

**Comment by kumare3** [comment]
@tnsetting let me push out a potential fix, that should fix the flake, there is one place where it can cause a flake!

**Comment by kumare3** [comment]
@tnsetting i think this is the fix - https://github.com/flyteorg/flytepropeller/pull/237. Once I get a +1, will merge

**Review by kumare3 — APPROVED** [review]

**Pull request merged**